src/test/scala/users/storemanager/StoreManagerRepositoryTest.scala

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright © 2022-2023 by Pervasive Cats S.r.l.s.
+ *
+ * All Rights Reserved.
+ */
+
+package io.github.pervasivecats
+package users.storemanager
+
+import java.sql.DriverManager
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration.SECONDS
+
+import com.dimafeng.testcontainers.JdbcDatabaseContainer.CommonParams
+import com.dimafeng.testcontainers.PostgreSQLContainer
+import com.dimafeng.testcontainers.scalatest.TestContainerForAll
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import eu.timepit.refined.auto.given
+import org.scalatest.EitherValues.given
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers.*
+import org.testcontainers.utility.DockerImageName
+
+import users.user.valueobjects.{EncryptedPassword, PlainPassword, Username}
+import users.storemanager.Repository
+import users.storemanager.Repository.{StoreManagerAlreadyPresent, StoreManagerNotFound, OperationFailed}
+import users.storemanager.valueobjects.Store
+import users.storemanager.entities.StoreManager
+import users.ValidationError
+import users.user.services.PasswordAlgorithm
+
+class StoreManagerRepositoryTest extends AnyFunSpec with TestContainerForAll {
+
+  private val usernameString: String = "matteo"
+  private val storeID: Long = 10
+  private val newStoreID: Long = 99
+
+  private val username: Username = Username(usernameString).getOrElse(fail())
+  private val wrongUsername: Username = Username("nonmatteo").getOrElse(fail())
+  private val store: Store = Store(storeID).getOrElse(fail())
+  private val newStore: Store = Store(newStoreID).getOrElse(fail())
```

Comment:
Could be moved to its test to not generate confusion, if used in more than one test could be moved down to the nearest place before all tests that use it, these are not Java constants